homeassistant/components/unmanic/config_flow.py

```diff
@@ -0,0 +1,121 @@
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Initialize."""
+        self._errors = {}
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        self._errors = {}
+
+        if user_input is not None:
+            configuration_already_exists = await self._config_exists(
+                user_input[CONF_HOST],
+                user_input[CONF_PORT],
+                self.hass,
+            )
+            connection_valid = await self._test_connection(
+                user_input[CONF_HOST],
+                user_input[CONF_PORT],
+                user_input[CONF_SSL],
+                user_input[CONF_VERIFY_SSL],
+                user_input[CONF_TIMEOUT],
+            )
+            if connection_valid and not configuration_already_exists:
+                return self.async_create_entry(
+                    title="Unmanic (" + user_input[CONF_HOST] + ")", data=user_input
+                )
+            elif configuration_already_exists:
+                self._errors["base"] = "already_configured"
+            else:
+                self._errors["base"] = "cannot_connect"
+
+            return await self._show_config_form(user_input)
+
+        user_input = {}
+        user_input[CONF_PORT] = DEFAULT_PORT
+        user_input[CONF_SSL] = DEFAULT_SSL
+        user_input[CONF_VERIFY_SSL] = DEFAULT_VERIFY_SSL
+        user_input[CONF_TIMEOUT] = DEFAULT_TIMEOUT
+
+        return await self._show_config_form(user_input)
+
+    async def _show_config_form(self, user_input):
+        """Show the configuration form to edit location data."""
+
+        data_schema = vol.Schema(
+            {
+                vol.Required(CONF_HOST): str,
+                vol.Required(CONF_PORT, default=user_input[CONF_PORT]): int,
+                vol.Optional(CONF_SSL, default=user_input[CONF_SSL]): bool,
+                vol.Optional(
+                    CONF_VERIFY_SSL, default=user_input[CONF_VERIFY_SSL]
+                ): bool,
+                vol.Optional(CONF_TIMEOUT, default=user_input[CONF_TIMEOUT]): int,
+            }
+        )
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=data_schema,
+            errors=self._errors,
+        )
+
+    async def _config_exists(self, host, port, hass):
+        """Return true if configuration already exists."""
+        if DOMAIN in hass.data:
+            for entry in hass.data[DOMAIN]:
+                if (
+                    host == hass.data[DOMAIN][entry].config_entry.data[CONF_HOST]
+                    and port == hass.data[DOMAIN][entry].config_entry.data[CONF_PORT]
+                ):
+                    return True
+        return False
+
+    async def _test_connection(self, host, port, ssl, verify_ssl, timeout):
+        """Return true if credentials is valid."""
+
+        try:
+            session = async_create_clientsession(self.hass)
```

Comment:
Why do you create a new session where, while you reuse the Home Assistant session in `__init__`?

Reply:
How can the one in `__init__` be used? Doesn't `_test_connection()` get called before setting up the entry, then once the entry is setup, then `async_setup_entry()` is called where `session` is created?

Reply:
You use two different functions to get a (new) session. Create vs get session. You could use get in both implementations.

Reply:
Do you mean literally switch `session = async_create_clientsession(self.hass)` for `session = async_get_clientsession(self.hass)`? If so, that creates an error with "Failed to connect" which I believe is a generic exception from aiohttp, likely due to a missing session.

Reply:
I forgot to import `async_get_clientsession` and it somehow didn't raise an import error. Anyways, it's all set now 👍